src/js/ripple/amount.js

```diff
@@ -1,53 +1,55 @@
-var sjcl  = utils.sjcl;
-var bn    = sjcl.bn;
-
-var BigInteger = utils.jsbn.BigInteger;
-
-var UInt160  = require('./uint160').UInt160;
-var Seed     = require('./seed').Seed;
-var Currency = require('./currency').Currency;
-
-//
-// Amount class in the style of Java's BigInteger class
-// http://docs.oracle.com/javase/1.3/docs/api/java/math/BigInteger.html
-//
+var assert    = require('assert');
+var BigNumber = require('bignumber.js');
+var extend    = require('extend');
+var utils     = require('./utils');
+var UInt160   = require('./uint160').UInt160;
+var Seed      = require('./seed').Seed;
+var Currency  = require('./currency').Currency;
+
+utils.configureBigNumber(BigNumber);
+
+function setRoundingMode(roundingMode) {
+  var config = BigNumber.config();
+  var oldRoundingMode = config.ROUNDING_MODE;
+  config.ROUNDING_MODE = roundingMode;
+  BigNumber.config(config);
+  return oldRoundingMode;
+}
 
 function Amount() {
   // Json format:
   //  integer : XRP
   //  { 'value' : ..., 'currency' : ..., 'issuer' : ...}
 
-  this._value       = new BigInteger(); // NaN for bad value. Always positive.
-  this._offset      = 0; // Always 0 for XRP.
+  this._value       = new BigNumber(NaN);
   this._is_native   = true; // Default to XRP. Only valid if value is not NaN.
-  this._is_negative = false;
   this._currency    = new Currency();
   this._issuer      = new UInt160();
-};
+}
 
 var consts = {
   currency_xns:      0,
   currency_one:      1,
   xns_precision:     6,
 
-  // BigInteger values prefixed with bi_.
-  bi_5:              new BigInteger('5'),
-  bi_7:              new BigInteger('7'),
-  bi_10:             new BigInteger('10'),
-  bi_1e14:           new BigInteger(String(1e14)),
-  bi_1e16:           new BigInteger(String(1e16)),
-  bi_1e17:           new BigInteger(String(1e17)),
-  bi_1e32:           new BigInteger('100000000000000000000000000000000'),
-  bi_man_max_value:  new BigInteger('9999999999999999'),
-  bi_man_min_value:  new BigInteger('1000000000000000'),
-  bi_xns_max:        new BigInteger('9000000000000000000'), // Json wire limit.
-  bi_xns_min:        new BigInteger('-9000000000000000000'),// Json wire limit.
-  bi_xns_unit:       new BigInteger('1000000'),
+  // bi_ prefix refers to "big integer"
+  bi_5:              new BigNumber('5'),
+  bi_7:              new BigNumber('7'),
+  bi_10:             new BigNumber('10'),
+  bi_1e14:           new BigNumber(String(1e14)),
+  bi_1e16:           new BigNumber(String(1e16)),
+  bi_1e17:           new BigNumber(String(1e17)),
+  bi_1e32:           new BigNumber('100000000000000000000000000000000'),
+  bi_man_max_value:  new BigNumber('9999999999999999'),
+  bi_man_min_value:  new BigNumber('1000000000000000'),
+  bi_xns_max:        new BigNumber('9000000000000000000'), // Json wire limit.
+  bi_xns_min:        new BigNumber('-9000000000000000000'),// Json wire limit.
+  bi_xrp_max:        new BigNumber('9000000000000'),  // Json wire limit.
```

Comment:
Can you clarify what is wrong and what we want? This is a new field corresponding to bi_xns_max, but divided by 1000000 so that it is expressed in XRP rather than drops.